### DISCUSSION TITLE
404 error and default redirect with a custom 404

### DISCUSSION THREAD

**Discussion by Westley-Tan** [vercel/next.js#21099] [Help]
I am checking out the blog-example in next.js and tried to edit it. I have a custom 404 page. I move the [slug] under pages/ folder. At first when I went to a random path like "/df df", it broke with the `cannot getStaticprops` error, so I added a redirect to 404 in that function "if Next cannot find the `dfkdf.md`. so it is working well. But now, instead of redirecting to my custom 404, next always to the default build-in 404 FIRST before being redirected again to my custom 404.

This only happens when I enter a random path after "/" but works okay (redirected immediately to custom 404) under other nested routes, like "/asdfadf/asdfas" or "/asdf/erer/ceff".

Any idea now to disable redirecting to the build-in 404?

**Comment by Westley-Tan** [comment]
I found the error... because I try to use the 
`import ErrorPage from "next/error"` and point to code 404,in the component function...

**Reply by eviva2000** [reply]
Hello Westley, as you solved your problem I would like to know how you did that. I have defined a 404.js in the pages folder and when I type a random URL it redirects me to that 404 page but it is only working locally. when I deploy the project to the S3 bucket it doesn't render a 404 page after typing a random url.

**Reply by Westley-Tan** [reply]
fallback: false

**Reply by Westley-Tan** [reply]
fallback: true will NOT redirect in production mode, it works only on development mode according to the official documentation.

I cannot find the link now, you can check https://nextjs.org/docs/basic-features/data-fetching#fallback-pages. It's mentioned somewhere.

**Reply by willvlad** [reply]
I wonder if you are getting a 404 response or a 200 in your header response when you get to the 404 page? 
I've been struggling to make it return a 404 response instead of 200

**Comment by Westley-Tan** [comment]
I saw both 404 and 200. As long as it can transit to custom 404 that what matters. No need to spend too much time in a library. Reactjs is still the mainstream here.  I also suspect this feature can be changed in the future if enough people bring it up.

**Reply by willvlad** [reply]
I wouldn't care about 404, but today Google penalises sites which don't have a 404 response on a 404 page. That's not ideal for those of use using nextjs, but you don't want to argue with their search engine, you just want to give it what it wants :) the question is how

**Reply by Westley-Tan** [reply]
Then I think the build-in 404 should be good? It looks kinda nice actually. Clean and minimum.
